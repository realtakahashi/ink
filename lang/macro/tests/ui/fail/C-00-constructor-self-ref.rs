use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(constructor)]
        fn invalid_self_ref(&self) -> Self {
            Self {}
        }

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
