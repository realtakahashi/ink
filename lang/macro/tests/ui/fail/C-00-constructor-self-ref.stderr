error: #[ink(constructor)] functions must not have any kind of `self` receiver
  --> $DIR/C-00-constructor-self-ref.rs:10:29
   |
10 |         fn invalid_self_ref(&self) -> Self {
   |                             ^^^^^
